regen-network/regen-web
# feat: enter and display CFC metadata

## Description

Closes: https://github.com/regen-network/regen-registry/issues/1120


- Create flows to handle CFC: Project and Batch
- Display metadata for CFC credit class, projects, and batches




---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

Viewable on-chain entities (C45 is a copy of what will be C02)
1. Credit Class: https://deploy-preview-1232--regen-registry.netlify.app/credit-classes/C45
2. Project: https://deploy-preview-1232--regen-registry.netlify.app/projects/C45-007
3. Batch: https://deploy-preview-1232--regen-registry.netlify.app/credit-batches/C45-007-20210101-20220101-001

Optional: Create Project and Batch:
1. Go to dashboard -> Projects -> Create Project -> Choose Credit Class C45
2. skip to Metadata step
3. Copy paste from one of CFC projects [here](https://github.com/regen-network/regen-registry-standards/tree/main/ops/C02/project-metadata)
4. In metadata field, change `"regen:cfcProjectId"` value to something random (to avoid on-chain id collision)
5. Optional*: To get a better-looking map on the Project Page, edit the Location. Choosing a result from the input will populate more geo-data into the location metadata, resulting in a nicer map.
6. Optional*: Set a Project Developer in the Roles section
7. Review and submit to confirm keplr transaction
8. Create a Batch from this project. You can copy/paste batch metadata from [here](https://github.com/regen-network/regen-registry-standards/tree/main/ops/C02/credit-batch-metadata).

Regression:
https://deploy-preview-1232--regen-registry.netlify.app/projects/C01-001
https://deploy-preview-1232--regen-registry.netlify.app/credit-classes/C01
https://deploy-preview-1232--regen-registry.netlify.app/credit-batches/C01-001-19821212-20220728-031
https://deploy-preview-1232--regen-registry.netlify.app/projects/wilmot

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
